crates/hyperswitch_connectors/src/connectors/novalnet/transformers.rs

```diff
@@ -125,99 +321,687 @@ impl<F, T> TryFrom<ResponseRouterData<F, NovalnetPaymentsResponse, T, PaymentsRe
+    fn from(item: NovalnetTransactionStatus) -> Self {
+        match item {
+            NovalnetTransactionStatus::SUCCESS | NovalnetTransactionStatus::CONFIRMED => {
+                Self::Success
+            }
+            NovalnetTransactionStatus::PENDING => Self::Pending,
+            NovalnetTransactionStatus::FAILURE
+            | NovalnetTransactionStatus::ON_HOLD
+            | NovalnetTransactionStatus::DEACTIVATED
+            | NovalnetTransactionStatus::PROGRESS => Self::Failure,
+        }
+    }
+}
 
-#[allow(dead_code)]
-#[derive(Debug, Serialize, Default, Deserialize, Clone)]
-pub enum RefundStatus {
-    Succeeded,
-    Failed,
-    #[default]
-    Processing,
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct NovalnetRefundSyncResponse {
+    result: ResultData,
+    transaction: Option<NovalnetResponseTransactionData>,
 }
 
-impl From<RefundStatus> for enums::RefundStatus {
-    fn from(item: RefundStatus) -> Self {
-        match item {
-            RefundStatus::Succeeded => Self::Success,
-            RefundStatus::Failed => Self::Failure,
-            RefundStatus::Processing => Self::Pending,
-            //TODO: Review mapping
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct NovalnetRefundsTransactionData {
+    amount: u32,
+    date: Option<String>,
+    currency: String,
+    order_no: String,
+    payment_type: String,
+    refund: RefundData,
+    refunded_amount: u32,
+    status: NovalnetTransactionStatus,
+    status_code: u16,
+    test_mode: u8,
+    tid: Option<u64>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct RefundData {
+    amount: u32,
+    currency: String,
+    payment_type: String,
+    tid: Option<u64>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct NovalnetRefundResponse {
+    pub customer: Option<NovalnetResponseCustomer>,
+    pub merchant: Option<NovalnetResponseMerchant>,
+    pub result: ResultData,
+    pub transaction: Option<NovalnetRefundsTransactionData>,
+}
+
+impl TryFrom<RefundsResponseRouterData<Execute, NovalnetRefundResponse>>
+    for RefundsRouterData<Execute>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: RefundsResponseRouterData<Execute, NovalnetRefundResponse>,
+    ) -> Result<Self, Self::Error> {
+        match item.response.result.status {
+            NovalnetAPIStatus::SUCCESS => {
+                let refund_id = item
+                    .response
+                    .transaction
+                    .clone()
+                    .and_then(|data| data.tid.map(|tid| tid.to_string()))
+                    .ok_or(errors::ConnectorError::ResponseHandlingFailed)?;
+
+                let transaction_status = match item.response.transaction.clone() {
+                    Some(transaction) => Some(transaction.status),
+                    None => None,
+                }
+                .ok_or_else(missing_field_err("transaction status"))?;
```

Comment:
You can use `map` here.

```suggestion
                let transaction_status = item.response
                    .transaction
                    .map(|transaction| transaction.status)
                    .ok_or_else(missing_field_err("transaction status"))?;
```